charm/kafka/lib/charms/layer/kafka.py

```diff
@@ -61,8 +61,12 @@ def install(self, zk_units=[], log_dir='logs'):
 
         config = hookenv.config()
 
+        broker_id = unitdata.kv().get('kafka.broker_id')
```

Comment:
Should use create_or_get_brokerid. If storage not attached, use configured log_dir or default.

Reply:
I don't think we need to use unitdata. We can always read from the file.